src/requiredBy.js

```diff
@@ -0,0 +1,12 @@
+export default function requiredBy(requiredByPropName, propType, defaultValue = null) {
+  return function validator(props, propName, componentName, ...rest) {
+    if (props[requiredByPropName]) {
+      if (props[propName] === defaultValue || typeof props[propName] === 'undefined') {
+        return new TypeError(
+          `${componentName}: when ${requiredByPropName} is true, prop “${propName}” must be present.`,
+        );
+      }
+    }
+    return propType(props, propName, componentName, ...rest);
+  };
```

Comment:
ha, that is a good point.

Conceptually, no, you're right :-/ one of the conventions that all of the validators in this project follow, tho, is that they all have a `.isRequired` on them. Maybe `validator.isRequired = type.isRequired;` would be sufficient? altho then the `defaultValue` logic wouldn't apply.